staging/src/k8s.io/legacy-cloud-providers/openstack/openstack.go

```diff
@@ -261,59 +268,63 @@ func configFromEnv() (cfg Config, ok bool) {
 	return
 }
 
-func createKubernetesClient(kubeconfigPath string) (*kubernetes.Clientset, error) {
-	klog.Info("Creating kubernetes API client.")
-
-	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
-	if err != nil {
-		return nil, err
-	}
-	cfg.DisableCompression = true
-
-	client, err := kubernetes.NewForConfig(cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	v, err := client.Discovery().ServerVersion()
-	if err != nil {
-		return nil, err
-	}
-
-	klog.Infof("Kubernetes API client created, server version %s", fmt.Sprintf("v%v.%v", v.Major, v.Minor))
-	return client, nil
+// SetInformers initializes secret lister
+func (os *OpenStack) SetInformers(informerFactory informers.SharedInformerFactory) {
+	os.secretLister = informerFactory.Core().V1().Secrets().Lister()
 }
 
 // setConfigFromSecret allows setting up the config from k8s secret
-func setConfigFromSecret(cfg *Config) error {
-	secretName := cfg.Global.SecretName
-	secretNamespace := cfg.Global.SecretNamespace
-	kubeconfigPath := cfg.Global.KubeconfigPath
-
-	k8sClient, err := createKubernetesClient(kubeconfigPath)
-	if err != nil {
-		return fmt.Errorf("failed to get kubernetes client: %v", err)
+func (os *OpenStack) setConfigFromSecret() error {
+	if os.secretLister == nil {
+		return fmt.Errorf("secret lister is not initialized")
 	}
 
-	secret, err := k8sClient.CoreV1().Secrets(secretNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	secretName := os.secretName
+	secretNamespace := os.secretNamespace
+	secret, err := os.secretLister.Secrets(secretNamespace).Get(secretName)
 	if err != nil {
-		klog.Warningf("Cannot get secret %s in namespace %s. error: %q", secretName, secretNamespace, err)
+		klog.Errorf("Cannot get secret %s in namespace %s. error: %q", secretName, secretNamespace, err)
 		return err
 	}
 
+	cfg := &Config{}
```

Comment:
why have this here outside of the if clause if it is only used inside?

Reply:
Done